src/frontends/pytorch/src/transforms/index_loop_getitem_replacer.cpp

```diff
@@ -0,0 +1,154 @@
+    ov::matcher_pass_callback callback = [](pattern::Matcher& m) {
+        auto loop_op = ov::as_type_ptr<v5::Loop>(m.get_match_root());
+        bool check_len_input = false;
+        if (auto len_reduce = ov::as_type_ptr<v1::ReduceSum>(loop_op->input_value(0).get_node_shared_ptr())) {
+            if (auto len_slice = ov::as_type_ptr<v8::Slice>(len_reduce->input_value(0).get_node_shared_ptr())) {
+                if (auto len_shape_of = ov::as_type_ptr<v3::ShapeOf>(len_slice->input_value(0).get_node_shared_ptr())) {
+                    check_len_input = true;
+                }
+            }
+        }
+        if (!check_len_input)
+            return false;
+
+        std::shared_ptr<Node> chunk_op;
+        size_t chunk_idx = 0;
+        auto loop_inputs = loop_op->input_values();
+        for (size_t i = 1; i < loop_inputs.size(); i++) {
+            if (cast_fw_node(loop_inputs.at(i).get_node_shared_ptr(), "aten::chunk")) {
+                chunk_op = loop_inputs.at(i).get_node_shared_ptr();
+                chunk_idx = i;
+            }
+        }
+        if (!chunk_op)
+            return false;
+
+        auto body = loop_op->get_function();
+        size_t chunk_param_idx;
+        bool found_param_idx = false;
+        for (auto input_desc : loop_op->get_input_descriptions()) {
+            if (input_desc->m_input_index == chunk_idx) {
+                chunk_param_idx = input_desc->m_body_parameter_index;
+                found_param_idx = true;
+            }
+        }
+        if (!found_param_idx)
+            return false;
+
+        auto chunk_param = body->get_parameters().at(chunk_param_idx);
+        auto param_targets = chunk_param->get_output_target_inputs(0);
+        if (param_targets.size() != 1)
+            return false;
+
+        auto getitem = param_targets.begin()->get_node()->shared_from_this();
+        if (!ov::as_type_ptr<v8::Gather>(getitem))
+            return false;
+
+        auto dim = chunk_op->input_value(2);
+        if (!ov::as_type_ptr<v0::Constant>(dim.get_node_shared_ptr()))
+            return false;
+
+        // connect chunk input directly to loop
+        auto chunk_input = chunk_op->input_value(0);
+        chunk_op->output(0).replace(chunk_input);
+        // len(chunks) is number of iterations
+        auto chunks_outside = chunk_op->input_value(1);
+        loop_op->input_value(0).replace(chunks_outside);
+
+        auto chunk_counter = getitem->input_value(1);
+
+        auto tensor_0 = v0::Constant::create(element::i32, Shape{1}, {0});
+        auto one_1d = v0::Constant::create(element::i32, Shape{1}, {1});
+
+        auto input_shape = std::make_shared<v3::ShapeOf>(chunk_input, element::i32);
+        auto input_dimension = std::make_shared<v8::Gather>(input_shape, dim, tensor_0);
+        auto init_chunk_size = std::make_shared<v1::Divide>(input_dimension, chunks_outside, true);
+
+        // Add 1 if input is not evenly divisible by chunks
```

Comment:
I believe it's better to use ov Ceil operation here instead of the code below
or you can use Divide with `m_pythondiv="true"`, but in this case you need to use Negative operator twice, e.g.: 
auto out = Divide(input_dimension, Negative(chunks_outside), **true**)
auto res = Negative(out);

Reply:
disccussed offline, Ceiling can cause problems here.